src/inabox/inabox-iframe-messaging-client.js

```diff
@@ -51,13 +53,25 @@ function createIframeMessagingClient(win) {
   }
   iframeClient.setSentinel(sentinel || getRandom(win));
 
-  // Bet the top window is the scrollable window and loads host script.
-  // TODO(lannka,#9120):
-  // 1) check window ancestor origin, if the top window is in same origin,
-  // don't bother to use post messages.
-  // 2) broadcast the request
-  iframeClient.setHostWindow(win.top);
-  return iframeClient;
+  iframeClient.setBroadcastMode(true);
+  return new Promise(resolve => {
+    let hostWin = win;
+    let j = 0;
+    const unlisten = iframeClient.registerCallback(MessageType.HOST_RESPONSE,
+        message => {
+          iframeClient.setBroadcastMode(false);
+          iframeClient.setHostWindow(message['source']);
+          unlisten();
+          resolve(iframeClient);
+        });
+    while (hostWin != win.top && j < 10) {
+      hostWin = hostWin.parent;
+      iframeClient.setHostWindow(hostWin);
+      iframeClient./*OK*/sendMessage(
```

Comment:
why do we need the `/*OK*/` here?

Reply:
It's disallowed by default: https://travis-ci.org/ampproject/amphtml/jobs/506892830#L446